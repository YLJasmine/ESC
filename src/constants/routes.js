/*
Each route is a page, i.e. sign up page should be at localhost:3000/signup
*/

export const SIGN_UP = '/signup';
export const SIGN_IN = '/signin';
export const LANDING = '/';
export const HOME = '/home';
export const ACCOUNT = '/account';
export const PROF = '/professor';
//export const PASSWORD_FORGET = '/pw-forget';