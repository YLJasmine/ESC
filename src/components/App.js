import React from 'react';
import {  BrowserRouter as Router,
          Route,
        } from 'react-router-dom';
// allows user to navigate to diff routes using react router.
import Navigation from './Navigation';

// importing the routes so that the content changes
import LandingPage from './Landing';
import SignUpPage from './SignUp';
import SignInPage from './SignIn';
import HomePage from './Home';
import AccountPage from './Account';
import ProfPage from './Professor';

import * as routes from '../constants/routes';
import withAuthentication from './withAuthentication';

const App = () =>
  <Router>
    <div>
      <Navigation />

      <hr/>

      <Route exact path={routes.LANDING} component={() => <LandingPage />} />
      <Route exact path={routes.SIGN_UP} component={() => <SignUpPage />} />
      <Route exact path={routes.SIGN_IN} component={() => <SignInPage />} />
      <Route exact path={routes.HOME} component={() => <HomePage />} />
      <Route exact path={routes.ACCOUNT} component={() => <AccountPage />} />
      <Route exact path={routes.PROF} component={() => <ProfPage />} />
      </div>
  </Router>

export default withAuthentication(App);
