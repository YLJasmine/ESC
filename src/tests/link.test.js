import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Adapter from "enzyme-adapter-react-16";
import toJson from 'enzyme-to-json';
configure({ adapter: new Adapter() });

import LandingPage from '../components/Landing';
import HomePage from '../components/Home';
import App from '../components/App';
import SignInPage from '../components/SignIn';
import * as routes from '../constants/routes';




test('Landing components should render as expected', () => {
  const component =shallow(<LandingPage />);
  expect(component.contains('Landing')).toBeTruthy;
  const tree=toJson(component);
  expect(tree).toMatchSnapshot();
});

test('should render the Landing component when visited', () => {
  const wrapper =mount(
    <MemoryRouter initialEntries={['/']} initialIndex={0}>
      <App />
    </MemoryRouter>
  )

  expect(wrapper.find(SignInPage).length).toBe(0);
  expect(wrapper.find(LandingPage)).toHaveLength(1);
});

test('should render the SignIn component when visited', () => {
  const wrapper =mount(
    <MemoryRouter initialEntries={['/signin']}>
      <App />
    </MemoryRouter>
  )

  //expect(wrapper.find(LandingPage)).toHaveLength(0);
  expect(wrapper.contains(SignInPage)).toBe(true);
});
