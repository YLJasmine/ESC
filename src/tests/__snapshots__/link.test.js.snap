// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing components should render as expected 1`] = `
<div>
  <h1>
    Landing Page
  </h1>
</div>
`;
